urllib3/_collections.py

```diff
@@ -304,17 +304,17 @@ def items(self):
         return list(self.iteritems())
 
     @classmethod
-    def from_httplib(cls, message, duplicates=('set-cookie',)): # Python 2
+    def from_httplib(cls, message): # Python 2
         """Read headers from a Python 2 httplib message object."""
-        ret = cls(message.items())
-        # ret now contains only the last header line for each duplicate.
-        # Importing with all duplicates would be nice, but this would
-        # mean to repeat most of the raw parsing already done, when the
-        # message object was created. Extracting only the headers of interest 
-        # separately, the cookies, should be faster and requires less
-        # extra code.
-        for key in duplicates:
-            ret.discard(key)
-            for val in message.getheaders(key):
-                ret.add(key, val)
-            return ret
+        headers = []
+         
+        for line in message.headers:
+            if line.startswith((' ', '\t')):
```

Comment:
In case I missed something there in the py2 HTTPMessage, which would avoid reinventing the wheel and still being reasonable fast for importing the whole object, i.e. a oneway pass through the headers, I welcome any hints and well deserved bitchslaps!